### PR TITLE
Support OpenSSL 1.1

### PR DESCRIPTION
**Pull request by chrisburr**

### CODE REVIEW

File: src/ssl/gsi.c [hunk 1]
```diff
@@ -55,14 +55,15 @@ gsiCheckIssuedWrapper( X509_STORE_CTX * ctx, X509 * x, X509 * issuer )
...
         return 0;
 #endif
 
-    ctx->error = ret;
-    ctx->current_cert = x;
-    ctx->current_issuer = issuer;
-    return ctx->verify_cb( 0, ctx );
+    X509_STORE_CTX_set_error( ctx, ret );
+    X509_STORE_CTX_set_current_cert( ctx, x );
+    // FIXME: This doesn't work in OpenSSL 1.1.0+ and I'm not sure what it's doing
```
Comment (chrisburr) [inline src/ssl/gsi.c:64]: So I don't forget: This needs to be fixed before merging.

**Comment by fstagni** [comment]
We may end up merging this, but:
- this project (pyGSI) is not going to be maintained further: we will replace pyGSI with M2Crypto starting from DIRAC v7r0 (so this year)
- we would need to decide if to use a new release or not, and then if using it in DIRAC Externals or DIRACOS. Most probably not for the DIRAC Externals (which are also legacy), maybe for DIRACOS.

CC @chaen

**Comment by chrisburr** [comment]
My motivation for writing it is for conda-forge as I think it would be nice to make the LHCbDIRAC client available there so analysts can use it easily regardless of there host OS. The options I can see are:

1. Never merge this as include it as a patch when building for conda
2. Merge and make a new release that only supports OpenSSL 1.1+
3. Use preprocessor magic to make this support any OpenSSL version and make a new release (probably not justified given this will be replaced soon)
4. Wait until v7r0 is released

I'd favour 1 or 2 rather than waiting for 4 as I guess it will take a while even if it's planned for this year?

**Comment by chaen** [comment]
Well I understand the urge for it, but instead of going in all directions separately, maybe we could push together in teh same direction, and get v7r0 out quicklier ? :)
And btw, even with v7r0 and M2Crypto, Openssl 1.1+ will not work as of now. Openssl removed some useful hack (https://github.com/openssl/openssl/issues/8177), and M2Crypto lacks an interface to enable some option in OpenSSL. 
So we will need to do PRs on M2Crypto anyway

**Comment by fstagni** [comment]
We may even go for option 2, and then use the new tag only for conda-forge.

**Pull request closed without merging**